File: 1585-the-kth-factor-of-n/1585-the-kth-factor-of-n.java
class Solution {
    public int kthFactor(int n, int k) {
        // Create new ArrayList
        ArrayList<Integer> factor_list = new ArrayList<>();

        // Loop through all numbers from 1 to n to get the factors
        for(int num=1;num<=n;num++){
            if(n%num == 0)
            {
                factor_list.add(num);

                if(factor_list.size()==k){
                    return num;
                }
            }

        }

        return -1;

        
    }
}

File: 1585-the-kth-factor-of-n/1585-the-kth-factor-of-n.py
class Solution:
    def kthFactor(self, n: int, k: int) -> int:
        factor_list = []

        # Calculate factor by iterating through all numbers from 1 to n
        for num in range(1, n + 1):
            # if current number num is a factor of n, add to list
            if (n % num) == 0:
                factor_list.append(num)

                # If current number is kth , then return
                if len(factor_list) == k:
                    return num

        return -1